project/src/main/webapp/event-details.js

```diff
@@ -83,21 +83,37 @@ function getLinkForOpportunity(opportunityId) {
 }
 
 /**
- * Gets event details from database and fills out event page with details
+ * Gets event details from database with eventId and fills out event page with details
  */
-function getEventDetails() {
+async function getEventDetails() {
+    //make sign up link go to correct
   const queryString = window.location.search;
   const urlParams = new URLSearchParams(queryString);
-  fetch(
-      '/create-event?' +
-      new URLSearchParams({'eventId': urlParams.get('eventId')}))
-      .then((res) => (res.json())).then((data) => {
-        document.getElementById('name').innerHTML = data['name'];
-        document.getElementById('description').innerHTML = data['description'];
-        document.getElementById('date').innerHTML = `Date: 
-          ${data['date'].month}/
-            ${data['date'].dayOfMonth}/${data['date'].year}`;
-        document.getElementById('location').innerHTML =
-          `Location: ${data['location']}`;
-      });
+  const eventId = urlParams.get('eventId');
+
+  //Register for event
+  const registerBool = urlParams.get('register') ? urlParams.get('register'): "false";
```

Comment:
I'm a little confused by this section - isn't registerBool the same value as urlParams.get('register') here? If it's true, then registerBool = true. If it's false, then registerBool = "false"

Since I think it's only used once, I think you can one-line the whole thing and just go `if (urlParams.get('register'))`